packages/atlas/src/components/BenefitCard/BenefitCard.tsx

```diff
@@ -0,0 +1,121 @@
+import BN from 'bn.js'
+import { FC, ReactNode } from 'react'
+
+import { NumberFormat } from '@/components/NumberFormat'
+import { Text } from '@/components/Text'
+import { JOY_CURRENCY_TICKER } from '@/config/joystream'
+import { useMediaMatch } from '@/hooks/useMediaMatch'
+
+import {
+  ActionWrapper,
+  Content,
+  ListElement,
+  Pattern,
+  RewardWrapper,
+  StyledButton,
+  StyledJoyTokenIcon,
+  StyledList,
+  TitleWrapper,
+  TokenRewardWrapper,
+  Variant,
+  Wrapper,
+} from './BenefitCard.styles'
+
+export type BenefitCardProps = {
+  variant?: Variant
+  title: string
+  description?: string
+  steps?: ReactNode[]
+  actionButton?: {
+    text: string
+    onClick?: () => void
+  }
+  joyAmount: BN | number
+  dollarAmount: number
+}
+
+export const BenefitCard: FC<BenefitCardProps> = ({
+  variant = 'full',
+  title,
+  description,
+  steps,
+  actionButton,
+  joyAmount,
+  dollarAmount,
+}) => {
+  const smMatch = useMediaMatch('sm')
+  const fullVariant = variant === 'full'
+
+  const RewardAmount = () => (
+    <RewardWrapper isCompact={!fullVariant}>
+      {!!dollarAmount && (
+        <NumberFormat
+          as="p"
+          format="dollar"
+          variant={!smMatch ? 'h500' : 'h600'}
+          value={dollarAmount}
+          margin={{ right: dollarAmount && !fullVariant && !smMatch ? 2 : 0 }}
+        />
+      )}
+      <TokenRewardWrapper>
+        {(!smMatch && !fullVariant && dollarAmount) || (!fullVariant && dollarAmount) || fullVariant ? (
```

Comment:
can we store this boolean in a variable, for the sake of clarity? I don't know, isJoyTokenVisible?